### PR TITLE
Phlare: Switch to the new Pprof endpoint

### PR DESCRIPTION
**Pull request by cyriltovena**
**What is this feature?**

This changes the API call made to Phlare when requesting merge of stack traces to use the new pprof endpoint.

**Why do we need this feature?**

This add extra informations as opposed to before like line number and files.

**Who is this feature for?**

Phlare users.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

The new API is not yet released but should be soon once we cut 0.1.2 on the Phlare side.

### CODE REVIEW

**Comment by cyriltovena** [comment]
Sounds like a javascript test is broken related to the top table change @joey-grafana

File: pkg/tsdb/phlare/query.go [hunk 1]
```diff
@@ -124,114 +127,179 @@ func makeRequest(qm queryModel, query backend.DataQuery) *connect.Request[querie
...
+	current := &ProfileTree{
+		Self:  sample.Value[0],
+		Level: 0,
+	}
+	for len(locations) > 0 {
+		current.locationID = locations[0]
+		current.Value = sample.Value[0]
+		current.Level = len(locations)
+
+		lines := profile.Location[locations[0]-1].Line
```
Comment (aocenas) [inline pkg/tsdb/phlare/query.go:250]: Why do we adjust the index to `locations[0]-1`? On previous line we set `current.locationID = locations[0]` but then seem to add lines from parent location?

File: pkg/tsdb/phlare/query.go [hunk 2]
```diff
@@ -124,114 +127,179 @@ func makeRequest(qm queryModel, query backend.DataQuery) *connect.Request[querie
...
+		if len(lines) == 0 {
+			locations = locations[1:]
+			continue
+		}
+		// The leaf is at lines[len(lines)-1].
+		current.Function = &Function{
+			FunctionName: profile.StringTable[profile.Function[lines[len(lines)-1].FunctionId-1].Name],
```
Comment (aocenas) [inline pkg/tsdb/phlare/query.go:257]: Similarly here it seems like we are accessing different function ID than we should.

Comment (aocenas) [reply pkg/tsdb/phlare/query.go:257]: I see in tests we have `id = array index + 1` so this works. Is that some convention? Does not seem something that is mentioned in the pprof protobuf comments.

Comment (cyriltovena) [reply pkg/tsdb/phlare/query.go:257]: I can add a test if you want that marshal and unmarshal with pprof go tool to verify it works.

Comment (cyriltovena) [reply pkg/tsdb/phlare/query.go:257]: Correct.

**Review by aocenas — APPROVED** [review]
Looks good, let's merge and I can work on adding this to flame graph vis.

**Pull request merged**